packages/expo-notifications/ios/EXNotifications/Notifications/Emitter/EmitterModule.swift

```diff
@@ -9,28 +9,40 @@ let onDidReceiveNotificationResponse = "onDidReceiveNotificationResponse"
 let onDidClearNotificationResponse = "onDidClearNotificationResponse"
 
 open class EmitterModule: Module, NotificationDelegate {
+  private var lastNotificationResponse: UNNotificationResponse?
   public func definition() -> ModuleDefinition {
     Name("ExpoNotificationsEmitter")
 
     Events([onDidReceiveNotification, onDidReceiveNotificationResponse, onDidClearNotificationResponse])
 
-    OnStartObserving {
+    OnCreate {
       NotificationCenterManager.shared.addDelegate(self)
     }
 
-    OnStopObserving {
+    OnDestroy {
       NotificationCenterManager.shared.removeDelegate(self)
     }
 
     AsyncFunction("getLastNotificationResponseAsync") {() -> [String: Any]? in
-      if let lastResponse: UNNotificationResponse = NotificationCenterManager.shared.lastResponse {
+      if let lastResponse: UNNotificationResponse = lastNotificationResponse {
         return EXNotificationSerializer.serializedNotificationResponse(lastResponse)
       }
       return nil
     }
 
     AsyncFunction("clearLastNotificationResponseAsync") {
-      NotificationCenterManager.shared.lastResponse = nil
+      lastNotificationResponse = nil
+    }
+
+    Function("getLastNotificationResponse") {() -> [String: Any]? in
+      if let lastResponse: UNNotificationResponse = lastNotificationResponse {
+        return EXNotificationSerializer.serializedNotificationResponse(lastResponse)
```

Comment:
let's move `EXNotificationSerializer.serializedNotificationResponse(lastResponse)` to line 55 (inside `didReceive(_ response: UNNotificationResponse,`) and that way the sync getter will do the least amount of work needed at the time it's called